model/map/src/main/java/org/keycloak/models/map/realm/MapRealmAdapter.java

```diff
@@ -521,6 +524,14 @@ public void addRequiredCredential(String cred) {
         if (model == null) {
             throw new RuntimeException("Unknown credential type " + cred);
         }
+        if (entity.getRequiredCredentials() != null) {
+            Optional<MapRequiredCredentialEntity> existing = entity.getRequiredCredentials().stream()
+                    .filter(credential -> Objects.equals(model.getType(), credential.getType()))
+                    .findFirst();
+            if (existing.isPresent()) {
+                throw new ModelDuplicateException("A Required Credential with given type already exists.");
+            }
+        }
```

Comment:
@mhajas  This piece of code could be replaced by
```
if (getRequiredCredentialsStream().anyMatch(credential -> Objects.equals(model.getType(), credential.getType()))) {
    throw new ModelDuplicateException("A Required Credential with given type already exists.");
}
```

which is much more readable but `getRequiredCredentialsStream()` transforms all entities to models and I wanted to avoid that. 

Another option would be to add new method `Optional<MapRequiredCredentialEntity> getRequiredCredentialByType(String type);` to `MapRealmEntity`. 

WDYT?

Reply:
I like method `getRequiredCredentialByType(String type)`

Reply:
I like it as well, only in a separate PR to think it through.
For example of the questions - this pattern may be common to other places, should we unify? Should this method even be available in an `Entity`?

This PR should introduce only minimal changes, and that the suggestion above satisfies.

Reply:
so leaving it as is for now and I'll create new enhancement issue to address it, thanks

Reply:
Sorry, I was not clear. Please use the form you suggested above.

Reply:
Created https://github.com/keycloak/keycloak/issues/11603